src/test/java/seedu/address/model/StudyTrackerTest.java

```diff
@@ -122,6 +124,19 @@ public void getStudySpotList_modifyList_throwsUnsupportedOperationException() {
         assertThrows(UnsupportedOperationException.class, () -> studyTracker.getStudySpotList().remove(0));
     }
 
+    @Test
+    public void hashCode_test() {
+        StudyTracker anotherSt = new StudyTracker();
+        StudyTracker deckSt = new StudyTracker();
+
+        studyTracker.addStudySpot(STARBUCKS);
+        anotherSt.addStudySpot(STARBUCKS);
+        deckSt.addStudySpot(DECK);
+
+        assertEquals(anotherSt.hashCode(), studyTracker.hashCode());
+        assertNotEquals(deckSt.hashCode(), studyTracker.hashCode());
+    }
+
     /**
      * A stub ReadOnlyStudyTracker whose study spots list can violate interface constraints.
```

Comment:
setStudySpot method bottom line is not tested

`
if (editedStudySpot.isFavourite() && favouriteStudySpots.contains(target)) {
 favouriteStudySpots.setStudySpot(target, editedStudySpot);
 }
`



this can be done with:

```
StudyTracker st = new StudyTracker();
StudySpot favStarbucks = new StudySpotBuilder(STARBUCKS).withFavourite(true).build();
st.addStudySpot(favStarbucks);
st.addStudySpotToFavourites(favStarbucks);
StudySpot editedStarbucks = new StudySpotBuilder(STARBUCKS)
                .withAddress(VALID_ADDRESS_DECK).withTags(VALID_TAG_QUIET).withFavourite(true)
                .build()
st.setStudySpot(favStarbucks, editedStarbucks);
assertEquals(studyTracker.findStudySpot(STARBUCKS.getName()), editedStarbucks);

```